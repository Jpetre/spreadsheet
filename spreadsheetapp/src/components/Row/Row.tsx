import React from 'react';
import Cell from '../Cell/Cell';

type RowProps = {
  x: number;
  y: number;
};

const Row: React.FC<RowProps> = (props: RowProps) => {
  const cells = [];
  const y = props.y;
  for (let x = 0; x < props.x; x += 1) {
    cells.push(
      <Cell
        key={`${x}-${y}`}
        y={y}
        x={x}
      />,
    );
  };
  return (
    <div className='Row'>
      {cells}
    </div>
  );
};

export default Row;