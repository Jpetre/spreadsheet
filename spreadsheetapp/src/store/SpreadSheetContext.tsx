import React from 'react';
import { ColumnDataType } from '../../../types/types';

export const SpreadSheetContext = React.createContext({
  data: {},
  updateData: (data: ColumnDataType) => { }
});
