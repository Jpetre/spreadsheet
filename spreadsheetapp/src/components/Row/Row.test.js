import React from 'react';
import Row from './Row';
import renderer from 'react-test-renderer';
import { SpreadSheetContext } from '../../store/SpreadSheetContext';
import mockData from '../../mocks/data.json';

test('Row render correctly', () => {
  const component = renderer.create(
    <SpreadSheetContext.Provider
      value={{
        data: mockData
      }}
    >
      <Row 
        x={6} 
        y={1} 
      />
    </SpreadSheetContext.Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});