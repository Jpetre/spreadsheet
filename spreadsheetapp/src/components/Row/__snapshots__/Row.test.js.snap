// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row render correctly 1`] = `
<div
  className="Row"
>
  <div
    className="Cell Cell--info"
  >
    1
  </div>
  <div
    className="Cell"
    onClick={[Function]}
  >
    12
  </div>
  <div
    className="Cell"
    onClick={[Function]}
  >
    15
  </div>
  <div
    className="Cell"
    onClick={[Function]}
  >
    20
  </div>
  <div
    className="Cell"
    onClick={[Function]}
  >
    18
  </div>
  <div
    className="Cell"
    onClick={[Function]}
  >
    15
  </div>
</div>
`;
